### ISSUE TITLE
Source forest creation failed error on symbolically linked hidden .#filename.c type of file names

### ISSUE THREAD

**Issue by Barro** [bazelbuild/bazel#4075]
Please provide the following information. The more we know about your system and use case, the more easily and likely we can help.

### Description of the problem / feature request / question:

Bazel fails with following type of message when there are .c files that named like ".#filename.c" in the directory that are symbolic links (sometimes a second build attempt is needed):

```
ERROR: Source forest creation failed: /<path-to-cache>/execroot/__main__/.#tst.c (File exists)
```

These come up when you edit files with emacs and have not saved the files. Bazel should not fail when these files are not listed anywhere in its BUILD files.

### If possible, provide a minimal example to reproduce the problem:

Write following commands on your favorite *nix shell:

    echo "int main() { return 0; }" > tst.c
    touch WORKSPACE
    echo 'cc_binary(name="tst", srcs=["tst.c"])' > BUILD
    ln -s nothing ".#tst.c"
    bazel build :all
    bazel build :all  # Second build might be needed when Bazel does not fail for the first time.

This should lead into following build message on the second build attempt (first build does succeed):

```
INFO: Analysed target //:tst (0 packages loaded).
INFO: Found 1 target...
ERROR: Source forest creation failed: /<path-to-cache>/execroot/__main__/.#tst.c (File exists)
INFO: Elapsed time: 0.196s
FAILED: Build did NOT complete successfully
```

### Environment info
* Operating System:

Basically Debian testing amd64 version:

```
$ java -version
openjdk version "1.8.0_144"
OpenJDK Runtime Environment (build 1.8.0_144-8u144-b01-1-b01)
OpenJDK 64-Bit Server VM (build 25.144-b01, mixed mode)
$ lsb_release  -a
No LSB modules are available.
Distributor ID: Debian
Description:    Debian GNU/Linux testing (buster)
Release:        testing
Codename:       buster
```

Also happens on FreeBSD, so there is not that much operating system dependent behavior here.

* Bazel version (output of `bazel info release`):

    release 0.7.0

###  Have you found anything relevant by searching the web?

A previously closed bug report describing the issue but without giving means to reproduce it: https://github.com/bazelbuild/bazel/issues/1064

**Comment by jmillikin** [comment]
Related: https://github.com/bazelbuild/bazel/issues/4060

**Comment by jin** [comment]
cc @jmmv please triage this issue if it's still relevant.

**Comment by jmmv** [comment]
I haven't been able to reproduce this, and I tried as described in the original report and both with the sandboxed and standalone strategies. So I suspect this was fixed at some point.
